packages/vite/src/node/plugins/css.ts

```diff
@@ -191,9 +191,23 @@ export function cssPlugin(config: ResolvedConfig): Plugin {
         if (deps) {
           // record deps in the module graph so edits to @import css can trigger
           // main import to hot update
+
           const depModules = new Set(
-            [...deps].map((file) => moduleGraph.createFileOnlyEntry(file))
+            await Promise.all(
+              [...deps].map(async (file) => {
+                if (cssLangRE.test(file)) {
+                  return moduleGraph.createFileOnlyEntry(file)
+                }
+
+                const url = file.startsWith(config.root)
+                  ? file.slice(config.root.length)
+                  : path.posix.join(FS_PREFIX, file)
```

Comment:
Would you check if `fileToUrl` isn't what we need here? https://github.com/vitejs/vite/blob/18a22089df34c32daddf3b280c58f25cae73f722/packages/vite/src/node/plugins/asset.ts#L153
```suggestion
                const url = fileToUrl(file, config, this)
```
Or maybe `urlReplacer`. Better to use these to avoid issues with other OSs like windows also. This is what `fileToUrl` will end up calling in dev for reference
```js
function fileToDevUrl(id: string, config: ResolvedConfig) {
  let rtn: string
  if (checkPublicFile(id, config)) {
    // in public dir, keep the url as-is
    rtn = id
  } else if (id.startsWith(config.root)) {
    // in project root, infer short public path
    rtn = '/' + path.posix.relative(config.root, id)
  } else {
    // outside of project root, use absolute fs path
    // (this is special handled by the serve static middleware
    rtn = path.posix.join(FS_PREFIX + id)
  }
  return config.base + rtn.replace(/^\//, '')
}
```
Also, it would be great to have a test for this so we can use the CI to check that Windows and other versions of node are working correctly. It could be worked out in a future PR, maybe @OneNail you have some ideas for this too.

Reply:
done!